src/components/pages/Site/Site.js

```diff
@@ -126,10 +126,22 @@ const SiteForm = ({
   handleLongitudeChange,
 }) => {
   const handleLngLatSwap = () => {
-    handleLatitudeChange(formik.getFieldProps('longitude').value)
-    handleLongitudeChange(formik.getFieldProps('latitude').value)
+    const currentLatitude = formik.getFieldProps('latitude').value
+    const currentLongitude = formik.getFieldProps('longitude').value
+
+    handleLatitudeChange(currentLongitude)
+    handleLongitudeChange(currentLatitude)
+
+    formik.setFieldTouched('longitude', true)
+    formik.setFieldTouched('latitude', true)
   }
 
+  // hack to ensure formik values are updated before manually triggering validation
+  useEffect(() => {
+    formik.validateForm()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [formik.getFieldProps('latitude').value, formik.getFieldProps('longitude').value])
```

Comment:
Im not sure about the deps array. Not that there is a lint exception, but that we are calling a function as a dep array value. Im not sure how that will work interms of how the effect triggers. Might be safer with `formik.values.<fieldName>`

Reply:
oh good idea - well swap for `formik.values.<fieldName>` and test it!